rpc/backend/tx_info.go

```diff
@@ -40,11 +42,23 @@ func (b *Backend) GetTransactionByHash(txHash common.Hash) (*rpctypes.RPCTransac
 	}
 
 	// the `res.MsgIndex` is inferred from tx index, should be within the bound.
-	msg, ok := tx.GetMsgs()[res.MsgIndex].(*evmtypes.MsgEthereumTx)
-	if !ok {
+	if len(tx.GetMsgs()) > 1 {
+		return nil, errors.New("invalid ethereum tx, more than 1 message in transaction")
+	}
+
+	msg, isEth := tx.GetMsgs()[res.MsgIndex].(*evmtypes.MsgEthereumTx)
+	_, isBankSend := tx.GetMsgs()[res.MsgIndex].(*banktypes.MsgSend)
+	if !isEth && !isBankSend {
 		return nil, errors.New("invalid ethereum tx")
 	}
```

Comment:
probably makes sense to adjust the error message here too, could be confusing otherwise
```suggestion
	if !isEth && !isBankSend {
		return nil, errors.New("neither a valid ethereum tx or bank transfer")
	}
```